### PR TITLE
[Main][Xcode12] Add support for Xcode 12

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [submodule "external/macios-binaries"]
     path = external/macios-binaries
     url = ../../xamarin/macios-binaries
-    branch = main
+    branch = d16-8
```
[the remark this answers is not in the document]
Comment (dalexsoto) [reply .gitmodules:4]: @rolfbjarne We can't build binaries with Mac build disabled unless you think we should just merge branches in `macios-binaries` (d16-8 => main) I do not see another workaround

File: Make.config [hunk 2]
```diff
@@ -111,9 +111,9 @@ WATCHOS_NUGET_VERSION_NO_METADATA=$(WATCHOS_NUGET_VERSION)-$(NUGET_PRERELEASE_ID
 WATCHOS_NUGET_VERSION_FULL=$(WATCHOS_NUGET_VERSION_NO_METADATA)+$(NUGET_BUILD_METADATA)
 
 # Xcode version should have both a major and a minor version (even if the minor version is 0)
-XCODE_VERSION=11.6
```
Comment (spouliot) [inline Make.config:114]: `APIDIFF_REFERENCES=` (earlier, around line 34) should also be updated to point to XI 14.0 stable release

File: msbuild/Xamarin.Shared/Xamarin.Shared.targets [hunk 3]
```diff
@@ -200,6 +200,30 @@ Copyright (C) 2018 Microsoft. All rights reserved.
...
+				<Keychain>$(CodesignKeychain)</Keychain>
+				<DisableTimestamp>False</DisableTimestamp>
+				<DisableTimestamp Condition="'$(_SdkIsSimulator)' == 'true' Or '$(MmpDebug)' == 'true' Or '$(MtouchDebug)' == 'true'">True</DisableTimestamp>
+				<CodesignAllocate>$(_CodesignAllocate)</CodesignAllocate>
+				<ResourceRules>$(CodesignResourceRules)</ResourceRules>
+				<NativeExecutable>%(AdditionalAppExtensions.Filename)</NativeExecutable>
+			</_AppExtensionCodesignProperties> 
+		</ItemGroup>
+	</Target>
+
```
Comment (spouliot) [inline msbuild/Xamarin.Shared/Xamarin.Shared.targets:226]: IIRC that was already in `main` and backported into `xcode12`, c.c. @chamons 
are sure you're not duplicating the logic ?

File: .gitmodules (hunk 1 above)
[the remark this answers is not in the document]
Comment (rolfbjarne) [reply .gitmodules:4]: Yes, I'd say we merge `d16-8` into `main` in `macios-binaries`-

**Review by spouliot — CHANGES_REQUESTED** [review]

File: msbuild/Xamarin.Shared/Xamarin.Shared.targets (hunk 3 above)
Comment (mandel-macaque) [reply msbuild/Xamarin.Shared/Xamarin.Shared.targets:226]: You are correct, appears twice in the file, removing.

**Review by spouliot — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Review by whitneyschmidt — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**